### DISCUSSION TITLE
Better UX on login

### DISCUSSION THREAD

**Discussion by rocketeerbkw** [sventorben/keycloak-home-idp-discovery#447] [Ideas / New features]
I've tested the functionality for our use case and this plugin works well, thanks!

The biggest annoyance is the change in UX when logging in so that it now requires multiple steps: first enter your email, then optionally enter a password. I know this is a common SSO pattern, but just because it's common doesn't mean it's good. In our use case, we have a minority of users that will use SSO, but it forces all of our users to have a worse UX.

![image](https://github.com/user-attachments/assets/b49393af-b216-4e35-9ae8-1a9ace7e6ab2)

I found an issue about the same topic that has been closed https://github.com/sventorben/keycloak-home-idp-discovery/issues/231 but I can't tell if it's possible to change this UX without make changes to this plugin or not.

If plugin changes are required, I would like to collaborate on that. If they're not required, could you provide a little more guidance on how to configure this in keycloak? I can help getting that in the docs if desired.

**Comment by sventorben** [comment]
Hey Brandon,

thanks for the feature request.
How about the suggestion I made in https://github.com/sventorben/keycloak-home-idp-discovery/issues/231#issuecomment-1699692843 ?

Would that work for you?

Best
Sven-Torben

**Reply by rocketeerbkw** [reply]
I don't fully understand how that solves the problem? We're not using social links, and I don't necessarily want an "SSO button," I just found that someone else was complaining about the UX.

We use keycloak as part of an open source solution, all of our users and clients exist in one realm, not sure how a second realm can work.

**Reply by sventorben** [reply]
If you don't necessarily want a button, what is you suggestion then to improve the UX?

**Reply by rocketeerbkw** [reply]
One idea is to always show email/pass. When the email field is completed, use ajax to determine if SSO is an option and redirect, or show "login with sso" button, or hide the password field, etc.

But I'm also open to suggestions. Maybe someone has done actual user testing and research on the topic?
